File: match.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if not s:
            return True
        match = 0
        for c in t:
            if s[match] == c:
                match += 1
            if len(s) == match:
                return True
        return False

File: 392. Is Subsequence/match.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        if (s.length() == 0)
            return true;
        int match = 0;
        for (int i = 0; i < t.length(); i++) {
            if (s[match] == t[i])
                match += 1;
            if (match == s.length())
                return true;
        }

        return false;
    }
};